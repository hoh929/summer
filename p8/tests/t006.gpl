// Set up system variables
int window_width = 400;
int window_height = 500;
string window_title = "p8::t006, compare to t006.jpg";
int circle_x_increment = 10;
int circle_y_increment = 10;
int red = 12;
int green = 33;
int blue = 50;
int have_printed = 0;

// declare the animation handler functions
forward animation circle_animate(circle cur_circle);

circle my_circle(animation_block = circle_animate, x = window_width/2, y = window_height/2, radius = 30);

textbox txt1(text = "let this run until the ball stops", y = 30);
textbox txt2(text = "then compare window to t06.jpg", y = 10);

// animation handler for the circle
animation circle_animate(circle cur_circle)
{
    if (cur_circle.x != 70 || cur_circle.y != 420)
    {
        red = (red + 1) % 100;
        cur_circle.red = red/100.0;
        green = (green + 1) % 100;
        cur_circle.green = green/100.0;
        blue = (blue + 1) % 100;
        cur_circle.blue = blue/100.0;

	    // if circle has reached either the left or right, reverse its direction
	    if (cur_circle.x < 0 || cur_circle.x + 2 * cur_circle.radius > window_width)
		    circle_x_increment = -circle_x_increment;

	    // if circle has reached either the top or bottom, reverse its direction
	    if (cur_circle.y < 0 || cur_circle.y  + 2 * cur_circle.radius> window_height)
		    circle_y_increment = -circle_y_increment;

	    // on each step of the animation, move the circle
	    cur_circle.x += circle_x_increment;
	    cur_circle.y += circle_y_increment;
    }
    else if (have_printed == 0)
    {
      print("cur_circle.red = " + cur_circle.red);
      print("cur_circle.green = " + cur_circle.green);
      print("cur_circle.blue = " + cur_circle.blue);
      print("cur_circle.x = " + cur_circle.x);
      print("cur_circle.y = " + cur_circle.y);
      have_printed = 1;
    }
}
