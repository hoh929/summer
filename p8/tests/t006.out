gpl.cpp::main()
  input file(tests/t006.gpl)
  random seed(42)
  read_keypresses_from_standard_input(false)
  dump_pixels(true, file = results/t006.mypixels)
  symbol_table(true)
  graphics(true)

gpl.cpp::main() Calling yyparse()


gpl.cpp::main() after call to yyparse().


Printing the symbol table from main()
int blue = 50
animation_block circle_animate
int circle_x_increment = 10
int circle_y_increment = 10
circle cur_circle
{
    animation_block = NULL,
    blue = double(1),
    drawing_order = int(0),
    green = double(0),
    h = int(20),
    proximity = int(4),
    radius = int(10),
    red = double(0),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(20),
    x = int(0),
    y = int(0)
}
int green = 33
int have_printed = 0
circle my_circle
{
    animation_block = circle_animate,
    blue = double(1),
    drawing_order = int(0),
    green = double(0),
    h = int(60),
    proximity = int(4),
    radius = int(30),
    red = double(0),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(60),
    x = int(200),
    y = int(250)
}
int red = 12
textbox txt1
{
    animation_block = NULL,
    blue = double(0),
    drawing_order = int(0),
    green = double(0),
    h = int(11),
    proximity = int(4),
    red = double(0),
    size = double(0.1),
    space = int(10),
    text = string("let this run until the ball stops"),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(10),
    x = int(0),
    y = int(30)
}
textbox txt2
{
    animation_block = NULL,
    blue = double(0),
    drawing_order = int(0),
    green = double(0),
    h = int(11),
    proximity = int(4),
    red = double(0),
    size = double(0.1),
    space = int(10),
    text = string("then compare window to t06.jpg"),
    user_double = double(0),
    user_double2 = double(0),
    user_double3 = double(0),
    user_double4 = double(0),
    user_double5 = double(0),
    user_int = int(0),
    user_int2 = int(0),
    user_int3 = int(0),
    user_int4 = int(0),
    user_int5 = int(0),
    user_string = string(""),
    user_string2 = string(""),
    user_string3 = string(""),
    user_string4 = string(""),
    user_string5 = string(""),
    visible = int(1),
    w = int(10),
    x = int(0),
    y = int(10)
}
int window_height = 500
string window_title = "p8::t006, compare to t006.jpg"
int window_width = 400
gpl.cpp::main() Calling window->initialize().


gpl.cpp::main() Passing control to window->main_loop().
print[46]: cur_circle.red = 0.27
print[47]: cur_circle.green = 0.48
print[48]: cur_circle.blue = 0.65
print[49]: cur_circle.x = 70
print[50]: cur_circle.y = 420

User quit program.  Executing termination blocks
